### PR TITLE
Added limit to index version count

### PR DESCRIPTION
**Pull request by brandon-m-skinner**
## Description

Currently there is no limit to the number of index versions an indexable can have. This is a bad idea, so we're adding a limit of 2 per indexable.

Also needed to adjust a lot of the tests due to the added restriction.

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [x] This change works and has been tested on a Go sandbox.
- [x] This change has relevant unit tests (if applicable).
- [x] This change has relevant documentation additions / updates (if applicable).

## Steps to Test

1. Execute `lando wp vip-search index-versions add post` 3 or 4 times.
2. Apply PR.
3. Execute `lando wp vip-search index-versions add post`. You should get a WP_Error telling you you can't add any more.
4. Verify with `lando wp vip-search index-versions list post` that no other versions were added.
5. Delete all the index versions.
6. Execute `lando wp vip-search index-versions add post` until you get the warning from 3.
7. Verify you only have two versions with `lando wp vip-search index-versions list post`
8. Activate users feature and repeat steps with the user indexable.

### CODE REVIEW

File: search/includes/classes/class-versioning.php [hunk 1]
```diff
@@ -9,6 +9,10 @@
 
 class Versioning {
 	const INDEX_VERSIONS_OPTION = 'vip_search_index_versions';
+	/**
+	 * The maximum number of index versions that can exist for any indexable.
+	 */
+	const MAX_VERSIONS = 2;
```
Comment (mjangda) [inline search/includes/classes/class-versioning.php:15]: Minor note, but I would call this something more obvious like `MAX_NUMBER_OF_VERSIONS`. At a glance, `MAX_VERSIONS` could mean a number of different things and the more obvious naming avoids the ambiguity.

File: search/includes/classes/class-versioning.php [hunk 2]
```diff
@@ -360,6 +364,19 @@ public function add_version( Indexable $indexable ) {
 			return new WP_Error( 'unable-to-get-next-version', 'Unable to determine next index version' );
 		}
 
+		$current_version_count = count( $versions );
+		if ( ( $current_version_count + 1 ) > self::MAX_VERSIONS ) {
```
Comment (nickdaugherty) [inline search/includes/classes/class-versioning.php:368]:
```suggestion
		if ( ( $current_version_count ) >= self::MAX_VERSIONS ) {
```

Writing it this way might be more easibly grokable?

File: search/includes/classes/class-versioning.php (hunk 1 above)
Comment (brandon-m-skinner) [inline search/includes/classes/class-versioning.php:15]:
```suggestion
	const MAX_NUMBER_OF_VERSIONS = 2;
```

File: search/includes/classes/class-versioning.php [hunk 3]
```diff
@@ -360,6 +364,19 @@ public function add_version( Indexable $indexable ) {
...
 
+		$current_version_count = count( $versions );
+		if ( ( $current_version_count ) >= self::MAX_VERSIONS ) {
+			return new WP_Error(
+				'too-many-index-versions',
+				sprintf(
+					'Currently, %d versions exist for indexable %s. Only %d versions allowed per indexable.',
+					$current_version_count,
+					$indexable->slug,
+					self::MAX_VERSIONS
```
Comment (brandon-m-skinner) [inline search/includes/classes/class-versioning.php:375]:
```suggestion
					self::MAX_NUMBER_OF_VERSIONS
```

File: search/includes/classes/class-versioning.php [hunk 4]
```diff
@@ -360,6 +364,19 @@ public function add_version( Indexable $indexable ) {
 			return new WP_Error( 'unable-to-get-next-version', 'Unable to determine next index version' );
 		}
 
+		$current_version_count = count( $versions );
+		if ( ( $current_version_count ) >= self::MAX_VERSIONS ) {
```
Comment (brandon-m-skinner) [inline search/includes/classes/class-versioning.php:368]:
```suggestion
		if ( ( $current_version_count ) >= self::MAX_NUMBER_OF_VERSIONS ) {
```

**Review by brandon-m-skinner — COMMENTED** [review]
Switching from MAX_VERSIONS to MAX_NUMBER_OF_VERSIONS

File: search/includes/classes/class-versioning.php [hunk 5]
```diff
@@ -360,6 +364,19 @@ public function add_version( Indexable $indexable ) {
 			return new WP_Error( 'unable-to-get-next-version', 'Unable to determine next index version' );
 		}
 
+		$current_version_count = count( $versions );
+		if ( ( $current_version_count ) >= self::MAX_NUMBER_OF_VERSIONS ) {
```
Comment (brandon-m-skinner) [inline search/includes/classes/class-versioning.php:368]:
```suggestion
		if ( $current_version_count >= self::MAX_NUMBER_OF_VERSIONS ) {
```

File: search/includes/classes/class-versioning.php (hunk 1 above)
Comment (brandon-m-skinner) [reply search/includes/classes/class-versioning.php]: Resolved via GitHub UI

**Review by nickdaugherty — APPROVED** [review]

**Comment by brandon-m-skinner** [comment]
r1217-stacks

**Pull request merged**